test/transform_done_test.cpp

```diff
@@ -57,6 +59,24 @@ TEST(TransformDone, Smoke) {
   EXPECT_EQ(count, 1);
 }
 
+TEST(TransformDone, StayDone) {
+  timed_single_thread_context context;
+
+  auto scheduler = context.get_scheduler();
+
+  int count = 0;
+
+  sequence(
+    just_done()
+      | transform_done(
+          []{ return just(); }) 
+      | on(scheduler),
+    lazy([&]{ ++count; }))
+    | sync_wait();
+
+  EXPECT_EQ(count, 1);
+}
```

Comment:
This test doesn't necessarily test that `just_done()` sends the done signal, since it would pass even if it sent `set_value()`.

Maybe something like:
```
{
  std::optional<unit> result = just_done() | sync_wait();
  assert(!result.has_value());
}
{
  std::optional<int> result =
    just_done()
    | transform_done([] { return just(42); })
    | sync_wait();
  assert(result.has_value());
  assert(result.value() == 42);
}
```